compiler/next/include/chpl/uast/Label.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_LABEL_H
+#define CHPL_UAST_LABEL_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/queries/UniqueString.h"
+#include "chpl/uast/SimpleBlockLike.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a label. For example:
+
+  \rst
+  .. code-block:: chapel
+
+    for 0..15 {
+      while true do
+        if !(i % 2) then break print; else break end;
+      label print writeln(i);
+      label end;
+    }
+ 
+  \endrst
+
+*/
+class Label final : public SimpleBlockLike {
```

Comment:
I'm not seeing why this class should be SimpleBlockLike.

Consider:

```
label mylabel {
  break mylabel;
}
```

results in `error: can only label for-, while-do- and do-while-statements`.

Similarly, this results in the same error:

```
label mylabel {
  for i in 1..10 {
    break mylabel;
  }
}
```

So, the Label can only contain a single Expression, which will be a for / while-do / do-while loop, and the BlockStyle is not relevant here.

Reply:
I think it comes down to whether or not you want to attach semantic meaning to SimpleBlockLike. That was not my intent, and even if we were to do that I do not think you can treat all `SimpleBlockLike` things in the same way even were `Label` not to be included as a subclass.

So maybe that does lend value to "has-a-block" over "is-a-block".

Reply:
Should I adjust some combination of the grammar / parser actions so that Label has to return a `const Loop*`?

Reply:
Done, see 9334157